examples/nitro.c

```diff
@@ -95,6 +95,71 @@ bool parse_cmdline(int argc, char *const argv[], struct cmdline *cmdline)
     return false;
 }
 
+void *listen_enclave_app_output(void *opague)
+{
+    int ret, sock_fd, bytes_read;
+    char buffer[BUFSIZE];
+    struct sockaddr_vm addr;
+    struct timeval timeval;
+
+    sock_fd = socket(AF_VSOCK, SOCK_STREAM, 0);
+    if (sock_fd < 0) {
+        puts("unable to create host socket");
+        return (void *)-1;
+    }
+
+    bzero((char *) &addr, sizeof(struct sockaddr_vm));
+    addr.svm_family = AF_VSOCK;
+    addr.svm_cid = VMADDR_CID_ANY;
+    addr.svm_port = 8081;
+
+    memset(&timeval, 0, sizeof(struct timeval));
+    timeval.tv_sec = 5;
+
+    ret = setsockopt(sock_fd, AF_VSOCK, SO_VM_SOCKETS_CONNECT_TIMEOUT, (void *) &timeval, sizeof(struct timeval));
+    if (ret < 0) {
+        close(sock_fd);
+        puts("unable to set socket options");
+        return (void *)-1;
+    }
+
+    ret = bind(sock_fd, (struct sockaddr *) &addr, sizeof(addr));
+    if (ret < 0) {
+        close(sock_fd);
+        puts("unable to bind the host socket to the address");
+        exit(1);
```

Comment:
This is the only `exit(1)` call, the rest `return (void *)-1`. Is this on purpose? Why exit the process?